components/resolved_ts/src/resolver.rs

```diff
@@ -432,7 +432,12 @@ impl Resolver {
         // Find the min start ts.
         let min_lock = self.oldest_transaction();
         let has_lock = min_lock.is_some();
-        let min_txn_ts = min_lock.as_ref().map(|(ts, _)| *ts).unwrap_or(min_ts);
+        // min_txn_ts is min_lock.ts - 1, because we need to guarantee
```

Comment:
Would it be a problem if all value of the `min_lock.ts` comes from `start_ts` of transactions, and `min(all_transaction_start_ts) - 1` is used to calculate which differs from before?

Reply:
I suppose the resolved-ts can be arbitrarily smaller. Setting a lower resolved-ts imposes a stricter constraint on resolved-ts users. The main inconvenience I can think of is attempting to map a resolved-ts to its corresponding transaction.

Reply:
The resolved-ts may decrease by 1 during TiKV upgrades. This might be a problem.